### PR TITLE
ko-327 Refactor test functions

### PR DESCRIPTION
**Pull request by johntrimble**

### CODE REVIEW

**Comment by bradfordcp** [comment]
From my perspective, this looks awesome. Nice job @johntrimble. I'm leaving a more in-depth review for the devs on the team.

File: mage/ginkgo/lib.go [hunk 1]
```diff
@@ -166,3 +178,111 @@ func (ns *NsWrapper) WaitForOutputContainsAndLog(description string, kcmd kubect
...
+func (ns *NsWrapper) EnableGossip(podName string) {
+	execArgs := []string{"-c", "cassandra",
+		"--", "bash", "-c",
+		"nodetool enablegossip",
+	}
+	k := kubectl.ExecOnPod(podName, execArgs...)
+	ns.ExecVPanic(k)
+}
+
+func (ns *NsWrapper) RetrieveDatacenterPodNames(dcName string) []string {
```
Comment (jimdickinson) [inline mage/ginkgo/lib.go:267]: I would go with
```suggestion
func (ns *NsWrapper) GetDatacenterPodNames(dcName string) []string {
```
to stick with the Kubernetes verbs

File: tests/config_change/config_change_suite_test.go [hunk 2]
```diff
@@ -58,57 +58,29 @@ var _ = Describe(testName, func() {
...
 			step = "creating a datacenter resource with 3 racks/3 nodes"
 			k = kubectl.ApplyFiles(dcYaml)
 			ns.ExecAndLog(step, k)
 
-			step = "checking the cassandra operator progress status is set to Ready"
-			json = "jsonpath={.status.cassandraOperatorProgress}"
-			k = kubectl.Get(dcResource).
-				FormatOutput(json)
-			ns.WaitForOutputAndLog(step, k, "Ready", 1800)
+			ns.WaitForDatacenterReady(dcName)
```
Comment (jimdickinson) [inline tests/config_change/config_change_suite_test.go:67]: 💯

File: tests/superuser-secret/superuser_secret_suite_test.go [hunk 3]
```diff
@@ -1,132 +0,0 @@
...
-	. "github.com/onsi/ginkgo"
-	. "github.com/onsi/gomega"
-
-	ginkgo_util "github.com/datastax/cass-operator/mage/ginkgo"
-	"github.com/datastax/cass-operator/mage/kubectl"
-)
-
-var (
-	testName          = "Superuser Secret"
-	namespace         = "test-superuser-secret"
```
Comment (jimdickinson) [inline tests/superuser-secret/superuser_secret_suite_test.go:21]: did this get duplicated into one of the above two tests?

Comment (johntrimble) [reply tests/superuser-secret/superuser_secret_suite_test.go:21]: Right, I should have made a comment about that. This was replaced by the other two superuser secret tests.

**Pull request merged**